assets/src/application/ui/layout/entityActionMenu/useEntityActionsManager.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+import { pathOr, assocPath } from 'ramda';
+
+import { EntityActionsManager, EntityMenuItems, MenuRegistry } from './types';
+
+let menuRegistry: MenuRegistry = {};
+
+const useEntityActionsManager = <EntityType extends string, MenuKey extends string>(
+	entityType: EntityType,
+	entityId: string
+): EntityActionsManager<MenuKey> => {
+	const [menuItems, setMenuItems] = useState([]);
+
+	const getMapKey = (itemKey: MenuKey) => `${entityType}:${entityId}:${itemKey}`;
+
+	const registerMenuItem = (key: MenuKey, render: () => React.ReactNode): void => {
+		const mapKey = getMapKey(key);
+		if (!menuItems.includes(mapKey)) {
+			setMenuItems((items) => [...items, mapKey]);
+			menuRegistry = assocPath([entityType, entityId, key], render(), menuRegistry);
```

Comment:
this `render()` is a bit confusing at first glance, a comment or an extraction would be helpful

Reply:
done